doc/source/dev/gitwash/development_setup.rst

```diff
@@ -0,0 +1,143 @@
+specifics for the SciPy_ project, and to suggest some default names.
+
+Set up and configure a github_ account
+======================================
+
+If you don't have a github_ account, go to the github_ page, and make one.
+
+You then need to configure your account to allow write access - see the
+``Generating SSH keys`` help on `github help`_.
+
+Create your own forked copy of SciPy_
+=========================================
+
+#. Log into your github_ account.
+#. Go to the SciPy_ github home at `SciPy github`_.
+#. Click on the *fork* button:
+
+   .. image:: forking_button.png
+
+   After a short pause, you should find yourself at the home page for
+   your own forked copy of SciPy_.
+
+.. include:: git_links.inc
+
+
+.. _set-up-fork:
+
+Set up your fork
+################
+
+First you follow the instructions for :ref:`forking`.
+
+Overview
+========
+
+::
+
+   git clone https://github.com/your-user-name/scipy.git
+   cd scipy
+   git remote add upstream https://github.com/scipy/scipy.git
+
+In detail
+=========
+
+Clone your fork
+---------------
+
+#. Clone your fork to the local computer with ``git clone
+   https://github.com/your-user-name/scipy.git``
+#. Investigate.  Change directory to your new repo: ``cd scipy``. Then
+   ``git branch -a`` to show you all branches.  You'll get something
+   like::
+
+      * master
+      remotes/origin/master
+
+   This tells you that you are currently on the ``master`` branch, and
+   that you also have a ``remote`` connection to ``origin/master``.
+   What remote repository is ``remote/origin``? Try ``git remote -v`` to
+   see the URLs for the remote.  They will point to your github_ fork.
+
+   Now you want to connect to the upstream `SciPy github`_ repository, so
+   you can merge in changes from trunk.
+
+.. _linking-to-upstream:
+
+Linking your repository to the upstream repo
+--------------------------------------------
+
+::
+
+   cd scipy
+   git remote add upstream https://github.com/scipy/scipy.git
+
+``upstream`` here is just the arbitrary name we're using to refer to the
+main SciPy_ repository at `SciPy github`_.
+
+Just for your own satisfaction, show yourself that you now have a new
+'remote', with ``git remote -v show``, giving you something like::
+
+   upstream	https://github.com/scipy/scipy.git (fetch)
+   upstream	https://github.com/scipy/scipy.git (push)
+   origin	https://github.com/your-user-name/scipy.git (fetch)
+   origin	https://github.com/your-user-name/scipy.git (push)
+
+To keep in sync with changes in SciPy, you want to set up your repository
+so it pulls from ``upstream`` by default.  This can be done with::
+
+   git config branch.master.remote upstream
+   git config branch.master.merge refs/heads/master
+
+You may also want to have easy access to all pull requests sent to the
+SciPy repository::
+
+   git config --add remote.upstream.fetch '+refs/pull/*/head:refs/remotes/upstream/pr/*'
```

Comment:
This is a ton of PRs, I would not recommend this. I've tried it before and it's actually a bit of a nightmare for something like SciPy.

Reply:
This whole Git tutorial was brought over from NumPy documentation. Mostly I just converted "NumPy" to "SciPy". I think it's an improvement to have it, but it needs a lot of attention. I'll get rid of this part, though.

Reply:
This used to be the best way to do it, but there's https://hub.github.com/ now which is way nicer.

Reply:
I didn't think just mentioning hub fit with the rest of the tutorial, so I removed this part entirely. I assumed this removes the line from `.git/config`:
```
fetch = +refs/pull/*/head:refs/remotes/upstream/pr/*
```